PR: https://github.com/odoo/odoo/pull/241864

From: a069536d5c9d015a829db5b0bf5b947bef8190f3
From: soza-odoo
Date: 2026-03-02 16:09:23

Breaking data model changes scores: del:2 + add:0, change matches:
-    peppol_activate_self_billing_sending = fields.Boolean(
-    peppol_self_billing_reception_journal_id = fields.Many2one(

Total Changes: 55

[IMP] account_peppol:  Improve the UX of Peppol Setting

In this commit:
* We remove the 'peppol config wizard'.
* Remove some deprecated fields from company model and related test cases.
* clean up dead code.

task-5438539

closes odoo/odoo#241864

Related: odoo/upgrade#9187
Signed-off-by: Sven Führ (svfu) <svfu@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account_peppol/models/res_company.py
+++ b/addons/account_peppol/models/res_company.py
@@ -98,23 +98,6 @@ class ResCompany(models.Model):
     peppol_metadata = fields.Json(string='Peppol Metadata')
     peppol_metadata_updated_at = fields.Datetime(string='Peppol meta updated at')
 
-    # Deprecated
-    peppol_activate_self_billing_sending = fields.Boolean(
-        string="Activate self-billing sending",
-        help="If activated, you will be able to send vendor bills as self-billed invoices via Peppol.",
-    )
-    # Deprecated
-    peppol_self_billing_reception_journal_id = fields.Many2one(
-        comodel_name='account.journal',
-        string='Self-Billing reception journal',
-        help="Any self-billed invoices / credit notes received via Peppol will be created in draft in this journal. Defaults to the first sale journal.",
-        domain=[('type', '=', 'sale')],
-        compute='_compute_peppol_self_billing_reception_journal_id',
-        store=True,
-        readonly=False,
-        inverse='_inverse_peppol_self_billing_reception_journal_id',
-    )
-
     # -------------------------------------------------------------------------
     # HELPER METHODS
     # -------------------------------------------------------------------------
@@ -275,28 +258,6 @@ class ResCompany(models.Model):
             journals_to_reset.is_peppol_journal = False
             company.peppol_purchase_journal_id.is_peppol_journal = True
 
-    @api.depends('account_peppol_proxy_state')
-    def _compute_peppol_self_billing_reception_journal_id(self):
-        for company in self:
-            if not company.peppol_self_billing_reception_journal_id and company.peppol_can_send:
-                company.peppol_self_billing_reception_journal_id = self.env['account.journal'].search([
-                    *self.env['account.journal']._check_company_domain(company),
-                    ('type', '=', 'sale'),
-                ], limit=1)
-                company.peppol_self_billing_reception_journal_id.is_peppol_journal = True
-
-    def _inverse_peppol_self_billing_reception_journal_id(self):
-        for company in self:
-            # This avoid having 2 or more sale journals from the same company with
-            # `is_peppol_journal` set to True (which could occur after changes).
-            journals_to_reset = self.env['account.journal'].search([
-                ('company_id', '=', company.id),
-                ('type', '=', 'sale'),
-                ('is_peppol_journal', '=', True),
-            ])
-            journals_to_reset.is_peppol_journal = False
-            company.peppol_self_billing_reception_journal_id.is_peppol_journal = True
-
     @api.depends('email')
     def _compute_account_peppol_contact_email(self):
         for company in self:

--- a/addons/account_peppol/models/res_config_settings.py
+++ b/addons/account_peppol/models/res_config_settings.py
@@ -104,16 +104,6 @@ class ResConfigSettings(models.TransientModel):
         registration_action = registration_wizard._action_open_peppol_form(reopen=False)
         return registration_action
 
-    # Deprecated
-    def button_open_peppol_config_wizard(self):
-        return {
-            'type': 'ir.actions.act_window',
-            'name': 'Advanced Peppol Configuration',
-            'res_model': 'peppol.config.wizard',
-            'view_mode': 'form',
-            'target': 'new',
-        }
-
     def button_peppol_disconnect_branch_from_parent(self):
         self.ensure_one()
         previous_parent_company_name = self.company_id.peppol_parent_company_id.name
@@ -129,12 +119,6 @@ class ResConfigSettings(models.TransientModel):
             }
         }
 
-    # Dreprecated
-    def button_peppol_register_sender_as_receiver(self):
-        """Register the existing user as a receiver."""
-        self.ensure_one()
-        return self.env['peppol.config.wizard'].new().button_peppol_register_sender_as_receiver()
-
     def button_reconnect_this_database(self):
         """Re-establish an out-of-sync connection"""
         self.ensure_one()
